backend/battle/serializers.py

```diff
@@ -19,6 +28,98 @@ class Meta:
+
+    battle = serializers.PrimaryKeyRelatedField(queryset=Battle.objects.all())
+    trainer = serializers.PrimaryKeyRelatedField(queryset=User.objects.all())
+
+    pokemon_1 = serializers.CharField(style={'base_template': 'textarea.html'}, required=False)
+    pokemon_2 = serializers.CharField(style={'base_template': 'textarea.html'}, required=False)
+    pokemon_3 = serializers.CharField(style={'base_template': 'textarea.html'}, required=False)
+
+    position_pkn_1 = serializers.ChoiceField(
+        choices=POSITION_CHOICES,
+        label="Position",
+        required=False)
+    position_pkn_2 = serializers.ChoiceField(
+        choices=POSITION_CHOICES,
+        label="Position",
+        required=False)
+    position_pkn_3 = serializers.ChoiceField(
+        choices=POSITION_CHOICES,
+        label="Position",
+        required=False)
+
+    class Meta:
+        model = Team
+        fields = (
+            "battle",
+            "trainer",
+            "position_pkn_1",
+            "position_pkn_2",
+            "position_pkn_3",
+            "pokemon_1",
+            "pokemon_2",
+            "pokemon_3"
+        )
+
+    def validate(self, attrs):
+        valid = verify_if_data_is_valid(attrs)
+        if valid is not True:
+            raise serializers.ValidationError(valid)
+        return attrs
+
+    def create(self, validated_data):
+
+        pokemon_1 = self.validated_data['pokemon_1']
+        pokemon_2 = self.validated_data['pokemon_2']
+        pokemon_3 = self.validated_data['pokemon_3']
+
+        pokemon_1_object = Pokemon.objects.get(name=pokemon_1)
+        pokemon_2_object = Pokemon.objects.get(name=pokemon_2)
+        pokemon_3_object = Pokemon.objects.get(name=pokemon_3)
+
+        position_pkn_1 = self.validated_data['position_pkn_1']
+        position_pkn_2 = self.validated_data['position_pkn_2']
+        position_pkn_3 = self.validated_data['position_pkn_3']
+
+        validated_data.pop('pokemon_1')
+        validated_data.pop('pokemon_2')
+        validated_data.pop('pokemon_3')
+        validated_data.pop('position_pkn_1')
+        validated_data.pop('position_pkn_2')
+        validated_data.pop('position_pkn_3')
+
+        instance = Team.objects.create(**validated_data)
+
+        PokemonTeam.objects.create(
+            team=instance,
+            pokemon=pokemon_1_object,
+            order=position_pkn_1)
+
+        PokemonTeam.objects.create(
+            team=instance,
+            pokemon=pokemon_2_object,
+            order=position_pkn_2)
+
+        PokemonTeam.objects.create(
+            team=instance,
+            pokemon=pokemon_3_object,
+            order=position_pkn_3)
+        try:
+            Team.objects.get(
+                trainer=validated_data['battle'].creator,
+                battle=validated_data['battle'])
+            Team.objects.get(
+                trainer=validated_data['battle'].opponent,
+                battle=validated_data['battle'])
```

Comment:
hmmmm, why is this required?

Reply:
I verify if exists the creator and opponent's teams